### PR TITLE
Completed Rails Forms Project

### PR DESCRIPTION
**Pull request by igakigongo**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  validates(:username, presence: true)
+  validates(:email, presence: true)
```
Comment (mcrd25) [inline app/models/user.rb]: you may even add some validation with a regular expression here [OPTIONAL]

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  validates(:password, presence: true)
```
Comment (mcrd25) [inline app/models/user.rb]: You may add a minimum length for the password if you like here [OPTIONAL]

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<h1>New User Form</h1>
+
+<%= form_for @user, url: {action: "create"} do |f| %>
```
Comment (mcrd25) [inline app/views/users/new.html.erb]: Good job with `form_for` implementation

However, you forgot to include commented out code to prove that you implemented it in the following ways:
- Regular HTML FORM
- using `form_tag` syntax

Please add back those implementations within comments. [REQUIRED]

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      puts "user saved"
```
Comment (mcrd25) [inline app/controllers/users_controller.rb]: You may use the rails `flash[:success]` hash instead. [OPTIONAL]

File: test/controllers/users_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
```
Comment (mcrd25) [inline test/controllers/users_controller_test.rb]: Great job implementing tests! 

You should definitely look into using RSpec next time! 👍

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (mcrd25) [reply app/controllers/users_controller.rb]: Also, this does nothing to give feedback to users but only the cli. Truly consider using the `flash` hash

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Great Job so far! Some work to be done before 🚀 

Hi @igakigongo and @keshav-c 👋,

## General Comments
- Great job implementing everything! You've mew almost ALL requirements and the form works well 👍 
- You forgot to set up your [linter]() for this project. Don't forget to do so in the next project! 😛 
- You have not implemented the [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) please remember to do so in the next project. If there are any concerns about how it works please reread [this link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) again. Below is a snippet of instructions for this project: 
Screen Shot 2019-08-14 at 8 42 42 AM
Basically, development acts as your main branch (sort of like master) and your feature branch is where you make most changes and you request your code reviews from your feature branch to development. hen project is completed you then merge development to master in another pull request.

## Required Changes
- Add commented out code for previous implementations of form in `new.html.erb`

## Additional Suggestions
- You should use `flash[:success]` or `flash[:error]` accordingly instead of `puts` statement in controllers for feedback messages
- You can improve the UI in the future if you feel like this is a project you'd have in your professional portfolio
- You may implement a form partial and reuse it for `new.html.erb` and `edit.html.erb`

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  def new
+    @user = User.new(username: "big good name")
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update_attributes(user_params)
+      puts "user updated"
+      redirect_to new_user_path
```
Comment (mcrd25) [inline app/controllers/users_controller.rb]: This should be redirected to `edit_user_path`. Please fix [REQUIRED]

**Comment by keshav-c** [comment]
I have committed the required changes. I will add visual styles to the project at a later date.

**Comment by azdrenymeri** [comment]
### Hello @keshav-c  and @igakigongo 
### It's a :+1:  from my side
#### You can merge it

File: app/views/users/new.html.erb (hunk 3 above)
Comment (keshav-c) [reply app/views/users/new.html.erb]: We left out commented code in accordance with Microverse instructions, and also you can see that code if you check the git log.

**Comment by keshav-c** [comment]
Fin

**Pull request merged**